src/cool_seq_tool/mappers/exon_genomic_coords.py

```diff
@@ -899,70 +910,32 @@ async def _genomic_to_tx_segment(
-            provided, will attempt to retrieve associated GRCh38 accession from UTA.
-        :return: Tuple containing GRCh38 accession, GRCh38 position, and error message
-            if unable to get GRCh38 representation
-        """
-        # Validate accession exists
-        if not grch38_ac:
-            grch38_ac = await self.uta_db.get_newest_assembly_ac(genomic_ac)
-            if not grch38_ac:
-                return None, None, f"Unrecognized genomic accession: {genomic_ac}."
-
-            grch38_ac = grch38_ac[0]
-
-        if grch38_ac != genomic_ac:
-            # Ensure genomic_ac is GRCh37
-            chromosome, _ = self.seqrepo_access.translate_identifier(
-                genomic_ac, Assembly.GRCH37.value
-            )
-            if not chromosome:
-                _logger.warning(
-                    "SeqRepo could not find associated %s assembly for genomic accession %s.",
-                    Assembly.GRCH37.value,
-                    genomic_ac,
-                )
-                return (
-                    None,
-                    None,
-                    f"`genomic_ac` must use {Assembly.GRCH37.value} or {Assembly.GRCH38.value} assembly.",
-                )
-            chromosome = chromosome[-1].split(":")[-1]
-            liftover_data = self.liftover.get_liftover(
-                chromosome, genomic_pos, Assembly.GRCH38
-            )
-            if liftover_data is None:
-                return (
-                    None,
-                    None,
-                    f"Lifting over {genomic_pos} on {genomic_ac} from {Assembly.GRCH37.value} to {Assembly.GRCH38.value} was unsuccessful.",
-                )
-            genomic_pos = liftover_data[1]
-            genomic_ac = grch38_ac
+    async def _get_grch38_pos(self, genomic_pos: int, genomic_ac: str) -> int | None:
+        """Liftover a GRCh37 coordinate to GRCh38
 
-        return genomic_ac, genomic_pos, None
+        :param genomic_pos: A genomic coordinate in GRCh37
+        :param genomic_ac: The genomic accession in GRCh38
+        :return The genomic coordinate in GRCh38
+        """
+        chromosome, _ = self.seqrepo_access.translate_identifier(
+            genomic_ac, target_namespaces=Assembly.GRCH38.value
+        )
+        chromosome = chromosome[-1].split(":")[-1]
+        liftover_data = self.liftover.get_liftover(
+            chromosome, genomic_pos, Assembly.GRCH38
+        )
+        if liftover_data is None:
+            return None
+        return liftover_data[1]
 
     async def _validate_gene_coordinates(
         self,
         pos: int,
         genomic_ac: str,
         gene: str,
     ) -> bool:
-        """Get gene given a genomic accession and position.
-
-        If multiple genes are found for a given ``pos`` and ``genomic_ac``, only one
-        gene will be returned.
+        """Validate that a genomic coordinate falls within the first and last exon
+            given a gene and accession
```

Comment:
I think this should be changed in the other PR

Reply:
is it ok to leave this check in `genomic_to_tx_segment` since this change would get merged into the issue-383 branch?

Reply:
You should do it in issue-383 since that's where you modified the method, right?

Reply:
Yup, it shouldn't show as a change once the other PR is merged + you pull in those changes